packages/docusaurus-types/src/index.d.ts

```diff
@@ -126,13 +126,15 @@ export type I18nConfig = {
   defaultLocale: string;
   locales: [string, ...string[]];
   localeConfigs: Record<string, Partial<I18nLocaleConfig>>;
+  queryString?: string;
 };
 
 export type I18n = {
   defaultLocale: string;
   locales: [string, ...string[]];
   currentLocale: string;
   localeConfigs: Record<string, I18nLocaleConfig>;
+  queryString?: string;
```

Comment:
I don't think `i18n` is a good place to add this.

Persisting a chosen locale in some storage is a theming concern, not a core concern. It's not all or nothing, and locale change links shouldn't necessarily have this querystring.

plus a `i18n.queryString` config is not very intuitive to make it global as is

I'd rather have this as a locale dropdown config for now